app/models/hbx_enrollment.rb

```diff
@@ -2848,6 +2854,69 @@ def continuous_coverage?
 
   private
 
+  def aptc_tax_household_enrollments
+    @aptc_tax_household_enrollments ||=
+      TaxHouseholdEnrollment.by_enrollment_id(id).select do |thh_enr|
+        thh_enr.tax_household_members_enrollment_members.where(
+          :family_member_id.in => thh_enr.tax_household.aptc_members.map(&:applicant_id)
+        ).present?
+      end
+  end
+
+  # Calculates sum of enrolled aptc member's of TaxHouseholdEnrollment ehb_premiums including Minimum Responsibility.
+  def sum_of_member_ehb_premiums(thh_enr)
+    aptc_family_member_ids = thh_enr.tax_household.aptc_members.map(&:applicant_id)
+    hbx_enrollment_members.where(:applicant_id.in => aptc_family_member_ids).reduce(0) do |sum, member|
+      sum + ivl_decorated_hbx_enrollment.member_ehb_premium(member)
+    end
+  end
+
+  def thh_enr_group_ehb_premium_of_aptc_members(thh_enrs)
+    thh_enrs.inject({}) do |premiums, thh_enr|
+      premiums[thh_enr] = { group_ehb_premium: float_fix(sum_of_member_ehb_premiums(thh_enr)) }
+      premiums
+    end
+  end
+
+  # Incase of SingleTaxHouseholdEnrollment,
+  #   updates TaxHouseholdEnrollment's applied_aptc with enrollment's applied_aptc_amount
+  def update_single_thh_enrollment(thh_enr, thh_enr_premiums)
+    thh_enr.update_attributes!(
+      {
+        applied_aptc: applied_aptc_amount,
+        group_ehb_premium: thh_enr_premiums[thh_enr][:group_ehb_premium]
+      }
+    )
+  end
+
+  # Incase of MultipleTaxHouseholdEnrollments, applied_aptc is either
+  #   1. group_ehb_premium (or)
+  #   2. thh_enr.available_max_aptc * elected_aptc_pct
+  # To make this inline with plan shopping logic we are considering tax_household_enrollment level ehb_premium.
+  def update_multiple_thh_enrollments(thh_enrs, thh_enr_premiums)
+    sum_of_group_ehb_premiums = thh_enr_premiums.values.collect { |val| val[:group_ehb_premium] }.sum
+
+    if applied_aptc_amount.to_f >= sum_of_group_ehb_premiums
```

Comment:
`>=` this doesn't seem right; In case of STHH you're updating `applied_aptc` with enrollment's `applied_aptc_amount` - this means, applied_aptc_amount is always the min of selected aptc & total ehb premium, which means - here, applied aptc amount will never be greater than sum of group ehb premiums.

And - This comparision might cause issues with float. Can we do this instead? - `enrollment.applied_aptc == enrollment.total_ehb_premium`